### ISSUE TITLE
DirectoryEntry.Name returns "" instead of the entity name

### ISSUE THREAD

**Issue by Vivihung** [Azure/azure-data-lake-store-net#7]
I'm calling GetDirectoryEntryAsync() with file path and directory path.
The **returned DirectoryEntry.Name is always empty string**.
Note: But DirectoryEntry.FullName returns the correct full path.

Based on the documentation, I would expect DirectoryEntry.Name provide the directory or filename in the storage.

Here is a sample code to repro
```C#
DirectoryEntry directoryEntry = await client.GetDirectoryEntryAsync("/local/vivihung-testadlssdk/Test.csv");
// DirectoryEntry directoryEntry = await client.GetDirectoryEntryAsync("/local/vivihung-testadlssdk");
Console.WriteLine($"Name: {directoryEntry.Name}");
Console.WriteLine($"Full Name: {directoryEntry.FullName}");
Console.WriteLine($"Last Modified Time: {directoryEntry.LastModifiedTime}");
Console.WriteLine($"Size: {directoryEntry.Length}");
```

Result
```
Name:
Full Name: /local/vivihung-testadlssdk/Test.csv
Last Modified Time: 1/10/2018 9:47:41 PM
Size: 15725240
```

**Comment by rahuldutta90** [comment]
Acknowledge it. Make the fix on sdk side so that GetFileStatus will return the directory/file name for name. Will roll out the change in the next release (probably next week).

**Comment by rahuldutta90** [comment]
Fixed and released 1.0.5
